routing/pathfind.go

```diff
@@ -1112,3 +1108,54 @@ func getProbabilityBasedDist(weight int64, probability float64,
 
 	return int64(dist)
 }
+
+// lastHopPayloadSize calculates the payload size of the final hop in a route.
+// It depends on the tlv types which are present and also whether the hop is
+// part of a blinded route or not.
+func lastHopPayloadSize(r *RestrictParams, finalHtlcExpiry int32,
+	amount lnwire.MilliSatoshi, legacy bool) uint64 {
+
+	if r.BlindedPayment != nil {
+		blindedPath := r.BlindedPayment.BlindedPath.BlindedHops
+		blindedPoint := r.BlindedPayment.BlindedPath.BlindingPoint
+
+		//nolint:lll
+		finalHop := route.Hop{
+			AmtToForward:     amount,
+			OutgoingTimeLock: uint32(finalHtlcExpiry),
+			LegacyPayload:    false,
+			EncryptedData:    blindedPath[len(blindedPath)-1].CipherText,
+		}
+		if len(blindedPath) == 1 {
+			finalHop.BlindingPoint = blindedPoint
+		}
+
+		// The final hop does not have a short chanID set.
+		return finalHop.PayloadSize(0)
+	}
+
+	var mpp *record.MPP
+	if r.PaymentAddr != nil {
+		mpp = record.NewMPP(amount, *r.PaymentAddr)
+	}
+
+	var amp *record.AMP
+	if r.Amp != nil {
```

Comment:
Would prefer putting this in a new commit since it's a bug fix for AMP - meanwhile don't think we can just encode it like this as the `childIndex` is encoded using `BigSize`, which means it can vary from 0 to 4 bytes. My suggestion is, if it's difficult to get the actual amp size here, maybe we could just define a const value `32+32+4`